### ISSUE TITLE
R6003 does not apply to "quick" top-level type definition

### ISSUE THREAD

**Issue by stdedos** [pylint-dev/pylint#8869]
### Bug description

For Python 3.9,

code
```py
# pylint: disable=consider-alternative-union-syntax
PlotlyColorScale = Union[list[str], list[tuple[Union[float, int], str]]]
```

Used as return type
```py
def _get_coloring() -> tuple[PlotlyColorScale, float | None]:
```

Suggestion https://pylint.readthedocs.io/en/stable/user_guide/messages/refactor/consider-alternative-union-syntax.html

interpreted as
```py
PlotlyColorScale: (list[str], list[tuple[(float, int), str]])
PlotlyColorScale = (list[str], list[tuple[(float, int), str]])
```

is not the same as the original.

Examples should be updated, or the warning silenced

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
Info - R6003: Consider using alternative Union syntax instead of 'Union'

in a.py:21
```

### Expected behavior

_Nothing_

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.5
Python 3.9.17 (main, Jun  6 2023, 20:11:04) 
[GCC 9.4.0]
```

### OS / Environment

Ubuntu 20.04

**Comment by DanielNoord** [comment]
I don't understand the issue. Why can't you use `|` here?

**Comment by stdedos** [comment]
Because it's not supported? 😓 

```patch
@@ -18,7 +18,7 @@
from a import b

-PlotlyColorScale = Union[list[str], list[tuple[Union[float, int], str]]]
+PlotlyColorScale = list[str] | list[tuple[float | int, str]]

@unique
```

```python
Traceback (most recent call last):
...
    PlotlyColorScale = list[str] | list[tuple[float | int, str]]
TypeError: unsupported operand type(s) for |: 'type' and 'type'
```

File does contain

```python
from __future__ import annotations
```

**Comment by DanielNoord** [comment]
Do you have a `pylintrc` or other configuration file?

This is supported on `3.10` so that is where my confusion came from. Either you have your `py-version` set to `3.10` or we should fix this when pylint is running on `3.9`. I don't have a setup available currently to check whether this is a bug in `pylint`.

**Comment by stdedos** [comment]
Updated the top of the file. There is no `py-version` set, and the environment is a 3.9.

**Comment by stdedos** [comment]
Just for posterity: https://peps.python.org/pep-0562/

```
PEP 604 – Allow writing union types as X | Y
Author:
Philippe PRADOS <python at prados.fr>, Maggie Moss <maggiebmoss at gmail.com>
Sponsor:
Chris Angelico <rosuav at gmail.com>
BDFL-Delegate:
Guido van Rossum <guido at python.org>
Discussions-To:
[Typing-SIG list](https://mail.python.org/archives/list/typing-sig@python.org/)
Status:
Accepted
Type:
Standards Track
Topic:
[Typing](https://peps.python.org/topic/typing/)
Created:
28-Aug-2019
Python-Version:
3.10
Post-History:
28-Aug-2019, 05-Aug-2020
```
